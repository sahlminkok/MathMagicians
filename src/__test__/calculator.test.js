import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Calculator from '../components/calculator';

describe('Calculator', () => {
  // Jest: tests with snapshots
  test('Calculator Component renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Renders do some Math! text', () => {
    render(<Calculator />);
    expect(screen.getByText(/do some Math!/)).toBeInTheDocument();
  });
});
