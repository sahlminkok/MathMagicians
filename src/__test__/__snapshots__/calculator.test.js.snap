// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Calculator Component renders correctly 1`] = `
<section
  className="calc"
>
  <h2>
    Let's do some Math!
  </h2>
  <div
    className="calculator"
  >
    <div
      className="result darkGray"
    >
      0
    </div>
    <div
      className="buttons"
    >
      <button
        className="grayColor"
        data-testid="button"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="grayColor"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="grayColor"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="orangeColor"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="grayColor"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="grayColor"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="grayColor"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="orangeColor"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="grayColor"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="grayColor"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="grayColor"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="orangeColor"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="grayColor"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="grayColor"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="grayColor"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="orangeColor"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="twoTimesWidth grayColor"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="grayColor"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="orangeColor"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</section>
`;
